src/libs/Navigation/FreezeWrapper.tsx

```diff
@@ -1,31 +1,24 @@
 import {useIsFocused, useNavigation, useRoute} from '@react-navigation/native';
-import lodashFindIndex from 'lodash/findIndex';
-import PropTypes from 'prop-types';
 import React, {useEffect, useRef, useState} from 'react';
 import {Freeze} from 'react-freeze';
 import {InteractionManager} from 'react-native';
+import ChildrenProps from '@src/types/utils/ChildrenProps';
 
-const propTypes = {
+type FreezeWrapperProps = ChildrenProps & {
     /** Prop to disable freeze */
-    keepVisible: PropTypes.bool,
-    /** Children to wrap in FreezeWrapper. */
-    children: PropTypes.node.isRequired,
+    keepVisible: boolean;
 };
 
-const defaultProps = {
-    keepVisible: false,
-};
-
-function FreezeWrapper(props) {
+function FreezeWrapper({keepVisible = false, children}: FreezeWrapperProps) {
     const [isScreenBlurred, setIsScreenBlurred] = useState(false);
     // we need to know the screen index to determine if the screen can be frozen
-    const screenIndexRef = useRef(null);
+    const screenIndexRef = useRef<number | null>(null);
     const isFocused = useIsFocused();
     const navigation = useNavigation();
     const currentRoute = useRoute();
 
     useEffect(() => {
-        const index = lodashFindIndex(navigation.getState().routes, (route) => route.key === currentRoute.key);
+        const index = navigation.getState().routes.findIndex((route) => route.key === currentRoute.key);
```

Comment:
just want to make sure , navigation cant be undefined in that case?

Reply:
Looking at the `useNavigation` implementation, no it can't be undefined 👍